tools/rum-models-generator/Sources/CodeGeneration/Generate/JSONSchema.swift

```diff
@@ -210,11 +223,28 @@ internal class JSONSchema: Decodable {
             merge(with: schema)
         }
 
-        // TODO: RUM-4571 [iOS] Generate RUM models for Usage Telemetry
-        // With addition of the "usage telemetry" the RUM schema was updated with unsupported constructs.
-        // To avoid `🚧 Unimplemented: "Building `SwiftAssociatedTypeEnum` case label for JSONObject is not supported` failure
-        // here we explicitly ignore the "usage" sub-schema.
-        oneOf = oneOf?.filter { $0.ref != "telemetry/usage-schema.json" }
+        oneOf = oneOf?.compactMap { obj in
+            if obj.title == "TelemetryCommonFeaturesUsage" || obj.title == "TelemetryMobileFeaturesUsage" {
+                // transform telemetry spec to mobile compatible
+                guard let oneOf = obj.oneOf else {
+                    return obj
+                }
+
+                // promote `feature` property to `title` if `title` is missing
+                for one in oneOf {
+                    if one.title == nil {
+                        one.title = one.properties?["feature"]?.const?.value.description
+                    }
+                }
```

Comment:
question/ What do we use the `title` for? I'm concerned about this change as effectively it produces different JSON schema than was loaded from the file. I wish we could avoid adding this hack and solve the problem in a different way. Are there cleaner options?

Reply:
this is a problem with model itself, because title was never required. 

I have backfilled the titles here https://github.com/DataDog/rum-events-format/pull/221 but we can never be sure. During the codegen, it is common practice to transform the given specs to something that makes more sense, but under the limits and this change is not altering the schema but rather backfilling when certain property is not available such as title.

I would prefer to not have this and make title required but that will be breaking.

Reply:
I'm more concerned by inferring the title from a fixed field of the schema (`"feature"`) - and doing this only for certain schemas (`"TelemetryCommonFeaturesUsage" ||  "TelemetryMobileFeaturesUsage"`). I basically wonder if there could be a general-purpose fallback that can benefit any subschemas. Otherwise, we will break the abstraction by leaking particular schema details to generic schema parser.

Alternatively, we could make it into configuration of `JSONSchemaReader` and get rid of hardcoded strings here - but effectively this will only blurry the solution and likely make it harder to read.

Reply:
I would refrain to apply this across the board, atleast we know where the problem is and this fix actually solves it. If we encounter similar problem, then we can think to generalize it like using some block for specific type names.

additionally, if we get the schema updates merged in, we don't need this workaround anyway as long as people will follow the existing pattern and never skip title. My problem is, it is not enforced and good intentions not always work.

Reply:
> additionally, if we get the schema updates merged in, we don't need this workaround anyway as long as people will follow the existing pattern and never skip title. My problem is, it is not enforced and good intentions not always work.

I see 👍. So after https://github.com/DataDog/rum-events-format/pull/221 is merged maybe we need better error handling rather than this hack? If `title` is missing in any schema, this can result with clear error describing the problem and a fix that should happen in in `rum-events-format`.